Mathlib/Algebra/Category/Ring/Center.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2025 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang, Junyan Xu
+-/
+import Mathlib.CategoryTheory.Adjunction.Limits
+import Mathlib.Algebra.Category.ModuleCat.Abelian
+import Mathlib.Algebra.Category.ModuleCat.ChangeOfRings
+import Mathlib.Algebra.Small.Module
+
+/-!
+# A categorical description of the center of a ring
+
+In this file we prove that the center of a ring `R` is isomorphic to `End (𝟭 R-Mod)` the
+endomorphism ring of the identity functor on the category of `R`-modules. Consequently, the ring
+structure of a commutative ring is completely determined by its module category.
+
+## Main results
+
+- `Subring.centerEquivEndIdFunctor`: the center of a ring `R` is isomorphic to `End (𝟭 R-Mod)`.
+- `RingEquiv.ofModuleCatEquiv`: if two commutative rings have equivalent module categories, they are
+  isomorphic as rings.
+
+-/
+
+universe v v'
+
+variable (R : Type*) [Ring R]
+
+open CategoryTheory
+
+noncomputable abbrev test [Small.{v} R] : ModuleCat.{v} R ≌ ModuleCat.{v} (Shrink.{v} R) :=
+  ModuleCat.restrictScalarsEquivalenceOfRingEquiv (Shrink.ringEquiv R)
+
+variable (M : ModuleCat R)
+
+set_option synthInstance.maxHeartbeats 100000 in
```

Comment:
```suggestion
```

Reply:
the `synthInstance.maxHeartbeats` is needed here, my suspections is that for `ModuleCat.Hom.instModule` to fire, we need `SMulCommClass R (center R) M`, which lean is struggling to find